### ISSUE TITLE
nslookup container hostname not working (Win2019, gMSA)

### ISSUE THREAD

**Issue by marknitek** [moby/moby#38722]
**Description**

I'am running a Windows 2019 Container Container on Windows 2019 Host with a gMSA in a Transparent Network. 
The gMSA works fine (`nltest /parentdomain` works and `nltest /sc_verify` works too) and i can query users and have access to other resources.
The Problem is i cannot nslookup the container name. 
My Domain Admin tells me he does not have a valid dns entry either. 
Since this is Windows Server 2019 it should be possible to use a gMSA that does not match the hostname, but for this example i even tried it with matching names but it does not work either. 

Another guy told me that this does indeed work with Windows 2016. So is this an issue with Windows Server 2019? I have no Windows Server 2016 Host to verify that. 

**Steps to reproduce the issue:**
1. Create a gMSA
2. Create a transparent network
2. Run a Container with: `docker run --hostname testdns --network Transparent --security-opt "credentialspec=file://testdns.json" mcr.microsoft.com/windows/servercore:ltsc2019
3. Run `nslookup testdns` from any computer in the network

**Describe the results you received:**
nslookup fails

**Describe the results you expected:**
nslookup works

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:           18.09.0
 API version:       1.39
 Go version:        go1.10.3
 Git commit:        33a45cd0a2
 Built:             unknown-buildtime
 OS/Arch:           windows/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.24)
  Go version:       go1.10.3
  Git commit:       33a45cd0a2
  Built:            11/07/2018 00:24:12
  OS/Arch:          windows/amd64
  Experimental:     false
```

**Output of `docker info`:**

```
Containers: 4
 Running: 3
 Paused: 0
 Stopped: 1
Images: 30
Server Version: 18.09.0
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: ics l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd gelf json-file local logentries splunk syslog
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 17763 (17763.1.amd64fre.rs5_release.180914-1434)
Operating System: Windows Server 2019 Standard Version 1809 (OS Build 17763.168)
OSType: windows
Architecture: x86_64
CPUs: 6
Total Memory: 9.966GiB
Name: LOGCONT02
ID: 6BBI:D6BV:IHL3:JE7Z:SW7B:S62R:5SNG:HSET:3GVW:R6XR:JFHF:BVOB
Docker Root Dir: C:\ProgramData\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: API is accessible on http://0.0.0.0:2376 without encryption.
         Access to the remote API is equivalent to root access on the host. Refer
         to the 'Docker daemon attack surface' section in the documentation for
         more information: https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Windows Server 2019 VM On-Prem as Docker Host

**Comment by tfenster** [comment]
@marknitek This https://github.com/moby/moby/issues/38779 might be related although for me it does work if I use the same hostname and gMSA name

**Comment by marknitek** [comment]
Yes its the same issue. I dont know why its not working with the same name. Maybe i have to do some more tests.

**Comment by thaJeztah** [comment]
ping @dineshgovindasamy @pradipd PTAL

**Comment by daschott** [comment]
@marknitek did the suggestions in https://github.com/moby/moby/issues/38779#issuecomment-492754863 help clarify this at all?

**Comment by marknitek** [comment]
Well as @tfenster said, this scenario should be supported and we should be able to choose if we want the hostname to match gmsa or not. We realy need that.

About the issue in our AD where even matching hostname\gmsa does not work with nslookup i am out of ideas. 
Any hints are welcome

**Comment by daschott** [comment]
@rpsqrd FYI

**Comment by rpsqrd** [comment]
If you run `ipconfig /registerdns` in the container, does that help name resolution?
